lib/hyacinth/dynamic_fields_loader.rb

```diff
@@ -2,7 +2,57 @@
 
 module Hyacinth
   module DynamicFieldsLoader
-    def self.load_rights_fields!
+    # Loads fields with a serialized representation of fields. If a field is already present that field is updated with
+    # the new attributes given. If there's a problem loading a field an error will be raised and all changes will be
+    # rolled back. Optionally vocabularies can also be created if a dynamic field depends on it.
+    #
+    # @param field_definitions [Hash] serialized hash containing field definitions
+    #   Format should look like the following example. Each hash should contain all the fields required by that model.
+    #      {
+    #         dynamic_field_categories: [
+    #           {
+    #             display_label: 'Descriptive Metadata',
+    #             metadata_form: 'descriptive',
+    #             dynamic_field_groups: [
+    #               {
+    #                 string_key: 'name',
+    #                 display_label: 'Name',
+    #                 dynamic_field_groups: {
+    #                   string_key: 'role',
+    #                   dynamic_fields: [
+    #                     { string_key: 'term', field_type: 'controlled_term' }
+    #                   ]
+    #                 }
+    #               }
+    #             ]
+    #           }
+    #         ]
+    #       }
+    # @param [opts] options to use when generating fields
+    # @option opts [Boolean] :load_vocabularies option to load any vocabularies fields are dependent on
+    def self.load_fields!(field_definitions, opts = {})
```

Comment:
This also seems potentially useful for loading fields between Hyacinth instances (i.e. syncing prod fields to a new dev instance).

Reply:
Yep, that's what I was thinking!